tests/EdgeToEdge/Routes/UpdateDonorRouteTest.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\EdgeToEdge\Routes;
+
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Client;
+use WMDE\Fundraising\DonationContext\Domain\Model\Donation;
+use WMDE\Fundraising\DonationContext\Domain\Model\DonorName;
+use WMDE\Fundraising\DonationContext\Tests\Data\ValidDoctrineDonation;
+use WMDE\Fundraising\DonationContext\Tests\Data\ValidDonation;
+use WMDE\Fundraising\Frontend\Factories\FunFunFactory;
+use WMDE\Fundraising\Frontend\Tests\EdgeToEdge\WebRouteTestCase;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FixedTokenGenerator;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\OverridingCampaignConfigurationLoader;
+use WMDE\Fundraising\Store\DonationData;
+
+/**
+ * @license GNU GPL v2+
+ */
+class UpdateDonorRouteTest extends WebRouteTestCase {
+
+	private const PATH = 'donation/update';
+	const CORRECT_UPDATE_TOKEN = 'b5b249c8beefb986faf8d186a3f16e86ef509ab2';
+	const INVALID_UPDATE_TOKEN = '2ba905fe68e61f3a681d8faf689bfeeb8c942b5b';
+
+	public function testWhenCorrectPrivatePersonDataIsPosted_addressIsChanged(): void {
+		$this->createEnvironment(
+			[],
+			function ( Client $client, FunFunFactory $factory ): void {
+				$this->setDefaultSkin( $factory, 'cat17' );
+				$donation = $this->newStoredDonation( $factory );
+				$crawler = $this->performRequest(
+					$client,
+					$this->newPrivateDonorData(),
+					$donation->getId(),
+					self::CORRECT_UPDATE_TOKEN,
+					self::CORRECT_UPDATE_TOKEN
+				);
+				$response = $client->getResponse();
+				$this->assertTrue( $response->isSuccessful() );
+				$this->assertContains(
+					'Hans Wurst, Teststraße 123, 12345 Mönchengladbach',
```

Comment:
This can become brittle when the formatting changes (from comma-separated to smth. else, probably during the next ticket). But you can leave it for now.